docs/explanation/remote-build.rst

```diff
@@ -89,12 +89,20 @@ Current
    ``--platform`` and ``--build-for`` behave differently than they do for
    :ref:`lifecycle commands<reference-lifecycle-commands>`.
 
-``--platform`` or ``--build-for`` can only be provided when the ``platforms``
-or ``architectures`` keywords are not defined in the project metadata
-(`[12]`_).
+If the project file contains a ``platforms`` or ``architectures`` key, then
+``--platform`` or ``--build-for`` filter the build plan defined by that key.
+For more information about build plans and filtering, see
+:ref:`Build plans <build-plans>`.
```

Comment:
Doesn't this mean that:

```
platforms:
  shiny:
    build-on: [riscv64]
    build-far: [riscv64]
  mainframey:
    build-on: [s390x]
    build-for: [s390x]
```

will accept `snapcraft pack --platform=shiny` but not `snapcraft remote-build --platform=shiny`?

Reply:
I might need to make the docs more clear.

This paragraph means that:
```
platforms:
  shiny:
    build-on: [riscv64]
    build-far: [riscv64]
  mainframey:
    build-on: [s390x]
    build-for: [s390x]
```
will accept both `snapcraft pack --platform=shiny` and `snapcraft remote-build --platform=shiny`. `remote-build` behaves like lifecycle commands when `platforms` or `architectures` are defined. 

PS I'm assuming `build-far` was a typo/joke and not part of the question.

Reply:
Yeah it was a typo. I'll follow up in a meeting :-)